helix-term/src/commands.rs

```diff
@@ -5622,7 +5622,11 @@ fn select_textobject(cx: &mut Context, objtype: textobject::TextObject) {
 
     cx.on_next_key(move |cx, event| {
         cx.editor.autoinfo = None;
-        if let Some(ch) = event.char() {
+
+        if let Some(ch) = event
+            .char()
+            .or((event.code == KeyCode::Enter).then_some('\n'))
```

Comment:
Searching for `\n` is not exactly correct unless the document has LF line endings. This needs to search for `doc.line_ending.as_str()` instead. For a doc with CRLF line endings, selecting around `\n`s would exclude the opening `\r`s. The `surround` code needs to be refactored to accept `&str`s instead of `char`s in order to search for line endings